### DISCUSSION TITLE
Welcome to Hydro-SDK Discussions!

### DISCUSSION THREAD

**Discussion by chgibb** [hydro-sdk/hydro-sdk#390] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by braj065** [comment]
Hi. I am working on a flutter project to run on web and mobile.
Will we be able to import the plugins from pub.dev? Also will the same code work if we build for web?

**Reply by chgibb** [reply]
Hi @Braj65 ,

Thanks for checking out Hydro-SDK! 

Currently, packages from pub.dev are not supported directly but you can create your own bindings for your favourite packages to use from your Typescript code. There is some documentation on the topic here https://github.com/hydro-sdk/hydro-sdk/blob/master/bindings.md

Eventually,  SWID (https://hydro-sdk.io/docs/design-documents/swid) will allow for more convenient use of packages from pub.dev. 

Flutter web is currently not supported due to a bug in the Flutter engine. There's some discussion about Flutter web support here https://github.com/hydro-sdk/hydro-sdk/issues/595

Is Flutter web a requirement for you? There are ways to work around the bug from within Hydro-SDK and support Flutter web.

**Reply by braj065** [reply]
Thanks @chgibb

**Comment by nuxusr** [comment]
Is development on Linux possible or only on a Mac?  Pulled some of the example repos and I can't seem to get anything to compile -- wondering if its just wrong version of flutter installed or if its a platform issue

errors looks like Flutter/version issues but figured i'd make sure the platform was ok before digging in more
```
ERROR: Try correcting the name to the name of an existing getter, or defining a getter or field named 'subtitle'.
ERROR:     table["subtitle"] = vmObject.subtitle != null
```

**Reply by chgibb** [reply]
Hi @nuxusr ,

Thanks for checking out Hydro-SDK! 
Can you post the commands and full output that got you to that error as well as the output of `flutter doctor -v` please?

**Reply by nuxusr** [reply]
So this command works great `npx hydroc build --out-dir assets/ota --profile debug` but when I do `npx hydroc run` it seems to get stuck. When i do `flutter run` i get all sorts of errors including the one above.

```
Flutter (Channel stable, 2.10.3, on Manjaro Linux 5.15.25-1-MANJARO, locale en_US.UTF-8)
    • Flutter version 2.10.3 at /home/nuxusr/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (2 weeks ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 32.1.0-rc1)
    • Android SDK at /home/nuxusr/Android/Sdk
    • Platform android-32, build-tools 32.1.0-rc1
    • Java binary at: /opt/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = /opt/google/chrome/google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[✓] Android Studio (version 2021.1)
    • Android Studio at /opt/android-studio
    • Flutter plugin version 64.1.2
    • Dart plugin version 211.7798
    • Java version OpenJDK Runtime Environment (build 11.0.11+0-b60-7590822)

```

**Reply by chgibb** [reply]
From `flutter doctor -v`, it appears you are using Flutter 2.10.3. Unfortunately, Hydro-SDK is currently only compatible with Flutter 2.0.6. 

The work to make Hydro-SDK compatible with Flutter 2.10.3 (and future Flutter releases) is tracked at a high-level in the [Binding Generator project](https://github.com/hydro-sdk/hydro-sdk/projects/5) (and other sub-projects) and landed onto the [swid-flutter branch](https://github.com/hydro-sdk/hydro-sdk/tree/swid-flutter). That branch is not stable. I mention it only to illustrate the volume of work that is ongoing.

**Reply by nuxusr** [reply]
Cool I'll downgrade and give that a shot. Thanks!

**Comment by duttaoindril** [comment]
Is hydro abandoned?

**Reply by chgibb** [reply]
Hi @duttaoindril , thanks for checking out Hydro-SDK! Hydro-SDK is on hold for awhile.

**Reply by duttaoindril** [reply]
> Hi @duttaoindril , thanks for checking out Hydro-SDK! Hydro-SDK is on hold for awhile. 

Are there any resources to understand how hydro works from the bottom up to keep development ongoing?
